Removed adm-zip

dpgraham opened the pull request:
* Added a test for unzipFile (which uses adm-zip)
* Tested that it worked with adm-zip and then failed once adm-zip was taken out
* Replaced 'adm-zip' with zip.extractAllTo from appium-supprt

(reviewers @jlipps @imurchie)

imurchie commented on test/basedriver/helpers-specs.js:
```diff
@@ -15,4 +23,20 @@ describe('helpers', () => {
+    it('should unzip a .zip file', async () => {
+      const appContents = await fs.readFile(getFixture('FakeIOSApp.app'));
+      await unzipFile(getFixture('FakeIOSApp.app.zip'));
+      await fs.readFile(getFixture('FakeIOSApp.app')).should.eventually.deep.equal(appContents);
```
I don't understand what this is asserting. Is seems like it is checking that the `FakeIOSApp.app` didn't change? Is the test just making sure `unzipFile` doesn't error out? If so, it should assert that so it is clear.

imurchie commented on test/basedriver/helpers-specs.js:
```diff
@@ -15,4 +23,20 @@ describe('helpers', () => {
+      const forceWindows = sinon.stub(system, 'isWindows', () => true);
+      const appContents = await fs.readFile(getFixture('FakeIOSApp.app'));
+      await unzipFile(getFixture('FakeIOSApp.app.zip'));
+      await fs.readFile(getFixture('FakeIOSApp.app')).should.eventually.deep.equal(appContents);
```
Again.

Maybe I'm misunderstanding how the `unzipFile` function works. Does it overwrite `FakeIOSApp.app`?

dpgraham commented:
@imurchie I got rid of the test that does not mock `system.isWindows` because that part of the code didn't change and it uses command line operations on the filesystem so it doesn't really work for unit-testing.

I updated the other test to use mock-fs so that it's all done in-memory and it doesn't touch the filesystem. I also changed it so that it hardcodes the base64 zip file into the mock fs, unzips it and then tests the contents.

imurchie commented on package.json:
```diff
@@ -67,8 +66,10 @@
     "eslint-plugin-babel": "^3.3.0",
     "eslint-plugin-import": "^1.10.2",
     "eslint-plugin-mocha": "^3.0.0",
+    "fancy-log": "^1.3.0",
```
I don't see this dep being used.

dpgraham commented:
Yeah I thought that the dependency was needed to fix an issue with `mock-fs` but it turns out I didn't need it. I tested without it (`rm -rf node_modules && npm install && gulp once`) and it built fine.

dpgraham commented:
Tagged and published to 2.7.2 (https://www.npmjs.com/package/appium-base-driver)
